framework/core/js/src/common/components/ModalManager.tsx

```diff
@@ -26,47 +26,62 @@ export default class ModalManager extends Component<IModalManagerAttrs> {
   // Keep track if there's an modal closing
   protected modalClosing: boolean = false;
 
+  protected keyUpListener: null | ((e: KeyboardEvent) => void) = null;
+
   view(vnode: Mithril.VnodeDOM<IModalManagerAttrs, this>): Mithril.Children {
-    return this.attrs.state.modalList.map((modal, i) => {
-      const Tag = modal?.componentClass;
-
-      return (
-        <div
-          key={modal.key}
-          class="ModalManager modal"
-          data-modal-key={modal.key}
-          data-modal-number={i}
-          role="dialog"
-          aria-modal="true"
-          style={{ '--modal-number': i }}
-          data-visibility-state={modal.animationState}
-          aria-hidden={this.attrs.state.modal !== modal && 'true'}
-        >
-          {!!Tag && (
-            <Tag
+    return (
+      <>
+        {this.attrs.state.modalList.map((modal, i) => {
+          const Tag = modal?.componentClass;
+
+          return (
+            <div
               key={modal.key}
-              {...modal.attrs}
-              animateShow={this.animateShow.bind(this)}
-              animateHide={this.animateHide.bind(this)}
-              state={this.attrs.state}
-            />
-          )}
-
-          <div class="backdrop" key={`backdrop-${modal.key}`} onclick={this.handleBackdropClick.bind(this)} />
-        </div>
-      );
-    });
+              class="ModalManager modal"
+              data-modal-key={modal.key}
+              data-modal-number={i}
+              role="dialog"
+              aria-modal="true"
+              style={{ '--modal-number': i }}
+              aria-hidden={this.attrs.state.modal !== modal && 'true'}
+            >
+              {!!Tag && (
+                <Tag
+                  key={modal.key}
+                  {...modal.attrs}
+                  animateShow={this.animateShow.bind(this)}
+                  animateHide={this.animateHide.bind(this)}
+                  state={this.attrs.state}
+                />
+              )}
+            </div>
+          );
+        })}
+
+        {this.attrs.state.backdropShown && (
+          <div
+            class="Modal-backdrop backdrop"
+            onclick={this.handlePossibleBackdropClick.bind(this)}
+            ontransitionend={this.onBackdropTransitionEnd.bind(this)}
+            data-showing={!!this.attrs.state.modalList.length}
```

Comment:
Ideally, the backdrop would show behind the latest open modal and above the others. I think a simple CSS trick would do it: (other comment below for follow up)

```suggestion
            data-showing={!!this.attrs.state.modalList.length}
            style={{'--modal-count': this.attrs.state.modalList.length}}
```